flutter_todo/lib/realm/realm_services.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter_todo/realm/schemas.dart';
+import 'package:realm/realm.dart';
+import 'package:flutter/material.dart';
+
+class RealmServices with ChangeNotifier {
+  static const String queryAllName = "getAllItemsSubscriptoion";
+  static const String queryMyItemsName = "getMyItemsSubscriptoion";
+
+  bool showAll = false;
+  bool offlineModeOn = false;
+  bool isWaiting = false;
+  late Realm realm;
+  User? currentUser;
+  App app;
+
+  RealmServices(this.app) {
+    if (app.currentUser != null || currentUser != app.currentUser) {
+      currentUser ??= app.currentUser;
+      realm = Realm(Configuration.flexibleSync(currentUser!, [Item.schema]));
+      showAll = (realm.subscriptions.findByName(queryAllName) != null);
+      if (realm.subscriptions.isEmpty) {
+        updateSubscriptions();
+      }
+    }
+  }
+
+  void updateSubscriptions() {
+    realm.subscriptions.update((mutableSubscriptions) {
+      mutableSubscriptions.removeByName(queryAllName);
+      mutableSubscriptions.removeByName(queryMyItemsName);
+      if (showAll) {
+        mutableSubscriptions.add(realm.all<Item>(), name: queryAllName);
+      } else {
+        mutableSubscriptions.add(realm.query<Item>(r'owner_id == $0', [currentUser?.id]), name: queryMyItemsName);
+      }
+    });
+  }
+
+  Future<void> sessionSwitch() async {
+    offlineModeOn = !offlineModeOn;
+    if (offlineModeOn) {
+      realm.syncSession.pause();
+    } else {
+      try {
+        isWaiting = true;
+        notifyListeners();
+        realm.syncSession.resume();
+        updateSubscriptions();
+        await realm.subscriptions.waitForSynchronization();
+      } finally {
+        isWaiting = false;
+      }
+    }
+    notifyListeners();
```

Comment:
They are causing the widgets to be rebuild if they are listening for the provided services. It refreshes the UI. Similar to setState. But this could be used also in a Stateless widget.